src/titiler/core/titiler/core/factory.py

```diff
@@ -495,19 +503,36 @@ def tile(
 
             tilesize = scale * 256
 
-            with Timer() as t:
-                with rasterio.Env(**self.gdal_config):
-                    with self.reader(src_path, tms=tms, **reader_params) as src_dst:
-                        data = src_dst.tile(
-                            x,
-                            y,
-                            z,
-                            tilesize=tilesize,
-                            tile_buffer=tile_buffer,
-                            **layer_params,
-                            **dataset_params,
-                        )
-                        dst_colormap = getattr(src_dst, "colormap", None)
+            if aoi:
+                feature = get_feature(aoi)
+                with Timer() as t:
+                    with rasterio.Env(**self.gdal_config):
+                        with self.reader(src_path, tms=tms, **reader_params) as src_dst:
+                            data = src_dst.tile(
+                                x,
+                                y,
+                                z,
+                                tilesize=tilesize,
+                                aoi=feature,
+                                tile_buffer=tile_buffer,
+                                **layer_params,
+                                **dataset_params,
+                            )
+                            dst_colormap = getattr(src_dst, "colormap", None)
+            else:
+                with Timer() as t:
+                    with rasterio.Env(**self.gdal_config):
+                        with self.reader(src_path, tms=tms, **reader_params) as src_dst:
+                            data = src_dst.tile(
+                                x,
+                                y,
+                                z,
+                                tilesize=tilesize,
+                                tile_buffer=tile_buffer,
+                                **layer_params,
+                                **dataset_params,
+                            )
+                            dst_colormap = getattr(src_dst, "colormap", None)
```

Comment:
Rather than all this copy/paste and deep if/else nesting, couldn't we unify things?:
```
feature = get_feature(aoi) if aoi else None
with Timer() as t:
    ...
    data = src_dst.tile(
        ...,
        aoi=feature,
        ...
    )
```

Reply:
Ah thank you! This was annoying me too: 22fd1073fd8483ae12a95acc7b61293b125af172